examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart

```diff
@@ -113,48 +113,58 @@ class TextFieldDemoState extends State<TextFieldDemo> {
         child: new ListView(
           padding: const EdgeInsets.symmetric(horizontal: 16.0),
           children: <Widget>[
-            new TextField(
-              icon: new Icon(Icons.person),
-              hintText: 'What do people call you?',
-              labelText: 'Name *',
-              onSaved: (InputValue val) { person.name = val.text; },
+            new TextFormField(
+              decoration: const InputDecoration(
+                icon: const Icon(Icons.person),
+                hintText: 'What do people call you?',
+                labelText: 'Name *',
+              ),
+              onSaved: (String value) { person.name = value; },
               validator: _validateName,
             ),
-            new TextField(
-              icon: new Icon(Icons.phone),
-              hintText: 'Where can we reach you?',
-              labelText: 'Phone Number *',
+            new TextFormField(
+              decoration: const InputDecoration(
+                icon: const Icon(Icons.phone),
+                hintText: 'Where can we reach you?',
+                labelText: 'Phone Number *',
+              ),
               keyboardType: TextInputType.phone,
-              onSaved: (InputValue val) { person.phoneNumber = val.text; },
+              onSaved: (String value) { person.phoneNumber = value; },
               validator: _validatePhoneNumber,
             ),
-            new TextField(
-              hintText: 'Tell us about yourself',
-              labelText: 'Life story',
+            new TextFormField(
+              decoration: const InputDecoration(
+                hintText: 'Tell us about yourself',
+                labelText: 'Life story',
+              ),
               maxLines: 3,
             ),
             new Row(
               crossAxisAlignment: CrossAxisAlignment.start,
               children: <Widget>[
                 new Expanded(
-                  child: new TextField(
+                  child: new TextFormField(
                     key: _passwordFieldKey,
-                    hintText: 'How do you log in?',
-                    labelText: 'New Password *',
+                    decoration: const InputDecoration(
+                      hintText: 'How do you log in?',
+                      labelText: 'New Password *',
```

Comment:
According to the Material mocks, the * should be red when the field doesn't have a value (even when the label is another colour). We didn't support this before, but it's something worth bearing in mind. Maybe a reason to take a Text here or something.